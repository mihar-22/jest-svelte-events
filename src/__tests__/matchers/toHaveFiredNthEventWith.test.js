import '../../'
import { render, fireEvent } from '@testing-library/svelte'
import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredNthEventWith', () => {
  test('passes when nth event contains the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(component).toHaveFiredNthEventWith(1, 50)
  })

  test('fails when no events were fired', async () => {
    const { component } = render(Comp)
    listen(component, 'e2')
    expect(() => {
      expect(component).toHaveFiredNthEventWith(1, 50)
    }).toThrowErrorMatchingSnapshot()
  })

  test('fails when nth event fires without the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredNthEventWith(1, 100)
    }).toThrowErrorMatchingSnapshot()
  })

  test('prints out all received payloads on fail', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    await fireEvent.click(button)
    await fireEvent.click(button)
    expect(() => {
      expect(component).toHaveFiredNthEventWith(1, 100)
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredNthEventWith', () => {
  test('passes when nth event does not contain the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(component).not.toHaveFiredNthEventWith(1, 100)
  })

  test('passes when no events have been fired', async () => {
    const { component } = render(Comp)
    listen(component, 'e2')
    expect(component).not.toHaveFiredNthEventWith(1, 100)
  })

  test('fails when last event fires with the correct payload', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e2')
    const button = getByText('e2')
    await fireEvent.click(button)
    expect(() => {
      expect(component).not.toHaveFiredNthEventWith(1, 50)
    }).toThrowErrorMatchingSnapshot()
  })
})
