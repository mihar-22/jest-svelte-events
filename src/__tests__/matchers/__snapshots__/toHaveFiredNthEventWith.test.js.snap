// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredNthEventWith fails when last event fires with the correct payload 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredNthEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired with: [32m50[39m

Event: e2
Number of calls: [31m1[39m
Events stack: [31m[\\"e2\\"][39m"
`;

exports[`.toHaveFiredNthEventWith fails when no events were fired 1`] = `
"Expected: [32m50[39m
Received: [31m\\"no events\\"[39m"
`;

exports[`.toHaveFiredNthEventWith fails when nth event fires without the correct payload 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredNthEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m
Received: [31m50[39m

Event: e2
Number of calls: [31m1[39m
Events stack: [31m[\\"e2\\"][39m"
`;

exports[`.toHaveFiredNthEventWith prints out all received payloads on fail 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredNthEventWith[2m([22m[32mexpected[39m[2m)[22m

Expected: [32m100[39m
Received: 
	   0: [31m50[39m,
	   1: [31m50[39m,
	   2: [31m50[39m

Event: e2
Number of calls: [31m3[39m
Events stack: [31m[\\"e2\\", \\"e2\\", \\"e2\\"][39m"
`;
